src/basic/log.c

```diff
@@ -585,6 +586,9 @@ int log_dispatch_internal(
         if ((level & LOG_FACMASK) == 0)
                 level = log_facility | LOG_PRI(level);
 
+        if (open_when_needed)
+                log_open();
```

Comment:
Calling `log_open` after `enforce_user(non-root)` makes it impossible to connect to `/dev/kmsg` and `/dev/console`, so under certain circumstances all [detailed error messages](https://github.com/systemd/systemd/issues/5000) are completely lost.

Reply:
I'm sorry. The previous version of the code behaves in the same way, so that is not related to this PR.

Reply:
Well, but that doesn't really change with this patch. I mean, all this patch does is that it optionally opens the log channel implicitly instead of expecting execute.c to do that explicitly. 

I guess we have a number of conflicting goals here: we want that fds don't remain open longer than necessary. But in case we do privileged logging we also want to keep them open across the priv dropping logic... Not sure how we could ever resolve that...

One way to improve things would be to add an automatic fallback to journal logging if we are configured to use kmsg or console but can't for some reason... But this is fragile too, as the journal isn't available in early boot, hence we would have to conditionally this somehow (maybe depending on the setting of StandardError=journal?)... yuck!